bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -975,6 +1070,137 @@ private static boolean checkIsRecordType(BType sourceType, BRecordType targetTyp
         return true;
     }
 
+    private static boolean checkRecordBelongsToAnydataType(MapValue sourceVal, BRecordType recordType,
+                                                           List<TypePair> unresolvedTypes) {
+        BType targetType = BTypes.typeAnydata;
+        TypePair pair = new TypePair(recordType, targetType);
+        if (unresolvedTypes.contains(pair)) {
```

Comment:
I think `HashSet` is a better choice if we are searching in the `unresolvedTypes`.

Reply:
This is passed on to `checkIsType` and in general we seem to be using lists in `checkIsType` and `checkIsLikeType`. Shall we do this change separately? Can you please create an issue with the suggestion?

Reply:
Help add labels
https://github.com/ballerina-platform/ballerina-lang/issues/25293

Reply:
Issue: https://github.com/ballerina-platform/ballerina-lang/issues/25293